packages/files-ui/src/Components/Modules/FileBrowsers/views/FilesList.tsx

```diff
@@ -851,10 +912,88 @@ const FilesList = ({ isShared = false }: Props) => {
     selectionContainsAFolder
   ])
 
-  const onShare = useCallback((fileSystemItem: FileSystemItemType) => {
-    setSelectedItems([fileSystemItem])
-    handleOpenShareDialog()
-  }, [handleOpenShareDialog])
+  const [contextMenuPosition, setContextMenuPosition] = useState<ContextMenuPosition | null>(null)
+
+  const handleContextMenuOnBrowser = useCallback((e: React.MouseEvent) => {
+    if (!controls) return
+    e.preventDefault()
+    // reset selected files if context menu was open
+    setSelectedItems([])
+    setContextMenuPosition({
+      mouseX: e.clientX - 2,
+      mouseY: e.clientY - 4
+    })
+  }, [controls])
+
+  const handleContextMenuOnItem = useCallback((e: React.MouseEvent, item: FileSystemItemType) => {
+    e.preventDefault()
+    e.stopPropagation()
+    // only keep current item selected if not in selected
+    if (!selectedItems.includes(item)) {
+      setSelectedItems([item])
+    }
+    setContextMenuPosition({
+      mouseX: e.clientX - 2,
+      mouseY: e.clientY - 4
+    })
+  }, [selectedItems])
+
+  const itemFunctions = useMemo(() => ({
+    deleteFile: onDeleteFile,
+    downloadFile: onDownloadFile,
+    handleShare: onShare,
+    moveFile: onMoveFile,
+    recoverFile: onRecoverFile,
+    reportFile: onReportFile,
+    viewFolder: handleViewFolder,
+    showFileInfo: onShowFileInfo,
+    previewFile: onShowPreview,
+    editFile: onEditFile
+  }), [
+    handleViewFolder,
+    onDeleteFile,
+    onDownloadFile,
+    onMoveFile,
+    onRecoverFile,
+    onReportFile,
+    onShowFileInfo,
+    onShowPreview,
+    onShare,
+    onEditFile
+  ])
+
+  const contextMenuOptions: IMenuItem[] = useMemo(() => {
+    if (selectedItems.length > 1) {
+      // bulk operations
+      return bulkActions
+    } else if (selectedItems.length === 1) {
+      // single item operations
+      const inSharedFolder = bucket?.type === "share"
+      const item  = selectedItems[0]
+      return getItemMenuOptions({
+        file: item,
+        inSharedFolder,
+        itemOperations: getItemOperations(item.content_type),
+        menuIconClass: classes.menuIcon,
+        ...itemFunctions
+      })
+    } else {
+      return browserOptions
+    }
+  }, [
+    browserOptions,
+    selectedItems,
+    bucket,
+    classes.menuIcon,
+    getItemOperations,
+    itemFunctions,
+    bulkActions
+  ])
+
+  const anchorPosition = useMemo(() => contextMenuPosition ? {
+    top: contextMenuPosition.mouseY,
+    left: contextMenuPosition.mouseX
+  } : undefined, [contextMenuPosition])
```

Comment:
Nit with ternary. I didn't put the object on several lines to keep it short
```suggestion
  const anchorPosition = useMemo(() => contextMenuPosition
    ? { top: contextMenuPosition.mouseY, left: contextMenuPosition.mouseX}
    : undefined
, [contextMenuPosition])
```